src/btwa_api/app/api/endpoints.py

```diff
@@ -16,3 +20,24 @@ def register(guid: str,
         return True
     else:
         raise HTTPException(status_code=400, detail="GUID already exists")
+
+
+@router.post("/data")
+def data(file: UploadFile = File(...)):
+    # TODO: Check if the upload is allowed based on phone number
+
+    # TODO: Check format of the file
+
+    connection_string = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
+    container_name = os.getenv("AZURE_STORAGE_CONTAINER")
+
+    blob_service_client = BlobServiceClient.from_connection_string(connection_string)
+
+    # TODO: generate name based on phone number fetched by check
+    directory = random.randrange(0, 9, 1)  # Ideally should be a prefix of phone number
+    blob_file = "./data/" + str(directory) + "/" + file.filename
```

Comment:
# directory traversal by naive fastapi file upload functionality

* suspected vulnerability source

https://github.com/covid19cz/bt-tracing-webapp/blob/3bbf81cfb9f0fb4161543bb1a18e5f42daa87b29/src/btwa_api/app/api/endpoints.py#L38
```
    blob_file = "./data/" + str(directory) + "/" + file.filename
```


* UploadFile behavior tested with example from Fastapi documentation, same code used in bt-tracing-webapp data endpoint

https://fastapi.tiangolo.com/tutorial/request-files/#multiple-file-uploads

```
POST /uploadfiles/ HTTP/1.1
Host: xxx
Content-Type: multipart/form-data; boundary=---------------------------39811971614456337441350123651
Content-Length: 242
Connection: close

-----------------------------39811971614456337441350123651
Content-Disposition: form-data; name="files"; filename="../../../test"
Content-Type: application/octet-stream

123

-----------------------------39811971614456337441350123651--
```

```
HTTP/1.1 200 OK
date: Tue, 17 Mar 2020 13:00:38 GMT
server: uvicorn
content-length: 31
content-type: application/json

{"filenames":["../../../test"]}
```

## Advisory

* exploitability depends on further implementation and azure.storage.blob.BlobServiceClient behavior
* fully validate any data submitted by client

Reply:
Thanks, @bodik! This is a "dummy" PR for benchmark purposes. In the end, we will change the filename completely :)